apis/Google.Devtools.AspNet/Google.Devtools.AspNet.Tests/CloudTrace/SimpleManagedTracerTest.cs

```diff
@@ -0,0 +1,256 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Devtools.Cloudtrace.V1;
+using Xunit;
+using Moq;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Google.Protobuf.Collections;
+
+namespace Google.Devtools.AspNet.Tests.CloudTrace
+{
+    public class SimpleManagedTracerTest
+    {
+        private const string TraceId = "trace-id";
+        private const string ProjectId = "project-id";
+
+        private static readonly ITraceConsumer UnusedConsumer = (new Mock<ITraceConsumer>()).Object;
+        private static readonly Dictionary<string, string> EmptyDictionary = new Dictionary<string, string>();
+        private static readonly StackTrace EmptyStackTrace = new StackTrace();
+
+        private static Cloudtrace.V1.Trace CreateTrace(string projectId = ProjectId, string traceId = TraceId)
+        {
+            return new Cloudtrace.V1.Trace()
+            {
+                ProjectId = ProjectId,
+                TraceId = TraceId
+            };
+        }
+
+        private static bool IsValidSpan(TraceSpan span, string name)
+        {
+            return IsValidSpan(span, name, 0);
+        }
+
+        private static bool IsValidSpan(TraceSpan span, string name, ulong parentId)
+        {
+            return IsValidSpan(span, name, parentId, SpanKind.Unspecified);
+        }
+
+        private static bool IsValidSpan(TraceSpan span, string name, ulong parentId, SpanKind kind)
+        {
+            return span.SpanId != 0 &&
+                kind.Convert() == span.Kind &&
+                name.Equals(span.Name) &&
+                span.StartTime != null &&
+                parentId == span.ParentSpanId &&
+                span.EndTime != null &&
+                span.StartTime.ToDateTime() <= span.EndTime.ToDateTime();
+        }
+
+        private static bool IsValidAnnotation(TraceSpan span, Dictionary<string, string> annotaion)
+        {
+            MapField<string, string> labels = span.Labels;
+            if (labels.Count != annotaion.Count)
+            {
+                return false;
+            }
+
+            foreach (KeyValuePair<string, string> label in labels)
+            {
+                if (annotaion[label.Key] != label.Value)
+                {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        [Fact]
+        public void SingleSpan()
+        {
+            Mock<ITraceConsumer> mockConsumer = new Mock<ITraceConsumer>();
+            IManagedTracer tracer = SimpleManagedTracer.Create(mockConsumer.Object, CreateTrace());
+
+            mockConsumer.Setup(c => c.Receive(
+                Match.Create<Traces>(
+                    t => t.Traces_.Count == 1 &&
```

Comment:
I'd either put it all on one line, or indent the second and third lines to show they're still part of the lambda.